aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -97,18 +97,39 @@ def validate_calc_job(inputs: Any, ctx: PortNamespace) -> Optional[str]:  # pyli
     return None
 
 
+def validate_stash_options(stash_options: Any, _: Any) -> Optional[str]:
+    """Validate the ``stash`` options."""
+    from aiida.common.datastructures import StashMode
+
+    target_base = stash_options.get('target_base', None)
+    source_list = stash_options.get('source_list', None)
+    stash_mode = stash_options.get('mode', StashMode.COPY.value)
+
+    if not isinstance(target_base, str) or not os.path.isabs(target_base):
+        return f'`metadata.options.stash.target_base` should be an absolute filepath, got: {target_base}'
+
+    if not isinstance(source_list, list) or any(not isinstance(src, str) or os.path.isabs(src) for src in source_list):
+        return f'`metadata.options.stash.source_list` should be a list of relative filepaths, got: {source_list}'
+
+    try:
+        StashMode(stash_mode)
+    except ValueError:
+        return f'`metadata.options.stash.mode` should be a member of StashMode, got: {stash_mode}'
```

Comment:
Since the user needs to provide the enum, would it be useful to specify where to get it from? So `...should be a member of aiida.common.datastructures.StashMode., got:...`, which would also be consistent with the help message (although not sure why we use `aiida.common.datastructures.StashMode.` instead of something like `aiida.common.datastructures:StashMode`).